sqlserver/datadog_checks/sqlserver/sqlserver.py

```diff
@@ -725,6 +728,16 @@ def _check_connections_by_use_db(self):
                 # Switch DB back to MASTER
                 cursor.execute(SWITCH_DB_STATEMENT.format(self.connection.DEFAULT_DATABASE))
 
+    def get_databases(self):
+        engine_edition = self.static_info_cache.get(STATIC_INFO_ENGINE_EDITION)
+        if not is_azure_sql_database(engine_edition):
+            db_names = [d.name for d in self.databases] or [
```

Comment:
I was just checking to see how `self.databases` gets initialized with autodiscovery. It looks like it relies on this code being called https://github.com/DataDog/integrations-core/blob/master/sqlserver/datadog_checks/sqlserver/sqlserver.py#L386-L416

this happens to be called before `self._schemas.run_job_loop(self.tags)` is run, which is fine but I wonder if its safer to call this as part of the initialization of the check 

cc @lu-zhengda this can be a follow on task too

Reply:
I had a problem with similar field.  And then realised that a bunch of initialisations happen at the end of SQLServer constructor but before the check is ran. 

STATIC_INFO_ENGINE_EDITION is set by 
`self.check_initializations.append(self.set_resolved_hostname)`